proposals/0000-splice-imports.rst

```diff
@@ -0,0 +1,345 @@
+   imported modules must be compiled for both host and target.
+   By distinguishing imported modules used at compile time(i.e. not used at
+   runtime), we can require only those modules to be compiled for the host.
+   Similarly, by distinguishing imported modules used at runtime (i.e. not used at
+   compile time), we can require only those modules to be compiled for the
+   target. It can be very hard or impossible to make some packages available on
+   some cross-compile target platforms, so this change would significantly
+   improve the applicability of ``TemplateHaskell`` in these scenarios.
+
+
+Definitions
+-----------
+
+level
+  Each expression exists at a level. The level is increased by 1 when
+  inside a quote and decreased by 1 inside a splice. Therefore the level of
+  an expression can be calculated as the number of quotes surrounding an expression
+  subtract the number of splices. For example::
+
+    -- foo is at level 0
+    foo = $(let
+      -- bar is at level -1
+      bar = $(let
+        -- baz is at level -2
+        baz = [|
+        -- qux is at level -1
+          qux = [|
+            -- quux is at level 0
+            quux = [|
+              -- quuz is at level 1
+              quuz = 0
+            |]
+          |]
+        |] in baz
+      ) in bar
+    )
+
+
+top-level splice
+  A splice, where the body is at a negative level, an unadorned
+  declaration splice or a quasiquoter.
+
+
+home module
+  A module from the package that is currently being compiled.
+
+
+Proposed Change
+---------------
+
+The ``splice`` modifier indicates that a module's imports are available at compile time.
+The absence of the ``splice`` modifier indicates that a module's imports are available
+at runtime.
+
+Resolution of scopes (often called "renaming") is blind to whether or not an
+identifier was imported with ``splice``. This is important because it will allow
+GHC to emit errors advising the user to modify their import declarations.
+
+The typechecker will be modified to emit errors in the following cases:
+1. It is an error to reference a ``splice`` imported name from a negative
+   level, and it is an error to reference a non-``splice`` imported name from
+   a non-negative level.
+
+2. It is an error to use a name that is both ``splice``-imported and non-``splice``
+   imported. Note that in some cases we could disambiguate these names by
+   inspecting the level at which they appeared, but we do not propose this.
+   See Ambiguity.
+
+Then,
+1. If a module is only available at compile time then the imports are only available in top-level splices.
+2. If a module is only available at runtime then the imports are not available in top-level splices.
+3. If a module is available at both runtime and compile time then the imports are available everywhere.
+
+The driver will be modified to ensure that, for modules with
+`-XTemplateHaskell`, object code is generated for ``splice`` imported modules,
```

Comment:
```suggestion
``-XTemplateHaskell``, object code is generated for ``splice`` imported modules,
```

I think you accidentally used Markdown instead of RST here.